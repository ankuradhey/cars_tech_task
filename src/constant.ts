export const constant = {
    host: "https://auto1-mock-server.herokuapp.com",
    carsUrl: "/api/cars",
    colorsUrl: "/api/colors",
    manufacturersUrl: "/api/manufacturers",
    carDetailsUrl: "/api/cars/{stockNumber}",
};
