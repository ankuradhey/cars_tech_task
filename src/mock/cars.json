{
    "cars": [
        {
            "stockNumber": 10019,
            "manufacturerName": "Fiat",
            "modelName": "Idea",
            "color": "blue",
            "mileage": {
                "number": 111955,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10084,
            "manufacturerName": "Skoda",
            "modelName": "Rapid",
            "color": "white",
            "mileage": {
                "number": 166819,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10176,
            "manufacturerName": "Mercedes-Benz",
            "modelName": "Sprinter",
            "color": "silver",
            "mileage": {
                "number": 125331,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10217,
            "manufacturerName": "Mercedes-Benz",
            "modelName": "Viano",
            "color": "green",
            "mileage": {
                "number": 119173,
                "unit": "km"
            },
            "fuelType": "Diesel",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10239,
            "manufacturerName": "Mercedes-Benz",
            "modelName": "SL-Klasse",
            "color": "blue",
            "mileage": {
                "number": 165176,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10283,
            "manufacturerName": "Chrysler",
            "modelName": "Crossfire",
            "color": "black",
            "mileage": {
                "number": 114177,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10385,
            "manufacturerName": "Porsche",
            "modelName": "718 Cayman",
            "color": "white",
            "mileage": {
                "number": 180397,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10477,
            "manufacturerName": "BMW",
            "modelName": "i3",
            "color": "blue",
            "mileage": {
                "number": 127584,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10481,
            "manufacturerName": "Volkswagen",
            "modelName": "Crafter",
            "color": "white",
            "mileage": {
                "number": 176629,
                "unit": "km"
            },
            "fuelType": "Diesel",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        },
        {
            "stockNumber": 10571,
            "manufacturerName": "Dodge",
            "modelName": "Nitro",
            "color": "yellow",
            "mileage": {
                "number": 115196,
                "unit": "km"
            },
            "fuelType": "Petrol",
            "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        }
    ],
    "totalPageCount": 100,
    "totalCarsCount": 1000
}