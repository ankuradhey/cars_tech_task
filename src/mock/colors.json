{"colors":["red","blue","green","black","yellow","white","silver"]}