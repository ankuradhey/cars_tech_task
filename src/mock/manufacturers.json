{
    "manufacturers": [
        {
            "name": "Audi",
            "models": [
                {
                    "name": "100"
                },
                {
                    "name": "200"
                },
                {
                    "name": "50"
                },
                {
                    "name": "80/90"
                },
                {
                    "name": "A1"
                },
                {
                    "name": "A2"
                },
                {
                    "name": "A3"
                },
                {
                    "name": "A4"
                },
                {
                    "name": "A4 Allroad"
                },
                {
                    "name": "A5"
                },
                {
                    "name": "A6"
                },
                {
                    "name": "A6 Allroad"
                },
                {
                    "name": "A7"
                },
                {
                    "name": "A8"
                },
                {
                    "name": "Allroad"
                },
                {
                    "name": "Cabriolet"
                },
                {
                    "name": "Coupe"
                },
                {
                    "name": "Q2"
                },
                {
                    "name": "Q3"
                },
                {
                    "name": "Q5"
                },
                {
                    "name": "Q7"
                },
                {
                    "name": "Quattro"
                },
                {
                    "name": "R8"
                },
                {
                    "name": "RS 3"
                },
                {
                    "name": "RS 4"
                },
                {
                    "name": "RS 5"
                },
                {
                    "name": "RS 6"
                },
                {
                    "name": "RS 7"
                },
                {
                    "name": "RS Q3"
                },
                {
                    "name": "S1"
                },
                {
                    "name": "S3"
                },
                {
                    "name": "S4"
                },
                {
                    "name": "S5"
                },
                {
                    "name": "S6"
                },
                {
                    "name": "S7"
                },
                {
                    "name": "S8"
                },
                {
                    "name": "SQ5"
                },
                {
                    "name": "SQ7"
                },
                {
                    "name": "TT"
                },
                {
                    "name": "TT Roadster"
                },
                {
                    "name": "V8"
                }
            ]
        },
        {
            "name": "BMW",
            "models": [
                {
                    "name": "1er"
                },
                {
                    "name": "2er"
                },
                {
                    "name": "3er"
                },
                {
                    "name": "4er"
                },
                {
                    "name": "5er"
                },
                {
                    "name": "6er"
                },
                {
                    "name": "7er"
                },
                {
                    "name": "8er"
                },
                {
                    "name": "X1"
                },
                {
                    "name": "X2"
                },
                {
                    "name": "X3"
                },
                {
                    "name": "X4"
                },
                {
                    "name": "X5"
                },
                {
                    "name": "X6"
                },
                {
                    "name": "Z1"
                },
                {
                    "name": "Z3"
                },
                {
                    "name": "Z4"
                },
                {
                    "name": "Z8"
                },
                {
                    "name": "i3"
                },
                {
                    "name": "i8"
                }
            ]
        },
        {
            "name": "Chrysler",
            "models": [
                {
                    "name": "300 C"
                },
                {
                    "name": "300 M"
                },
                {
                    "name": "Crossfire"
                },
                {
                    "name": "Daytona Shelby"
                },
                {
                    "name": "ES"
                },
                {
                    "name": "GS"
                },
                {
                    "name": "GTS"
                },
                {
                    "name": "Grand Voyager"
                },
                {
                    "name": "Le Baron"
                },
                {
                    "name": "Neon"
                },
                {
                    "name": "New Yorker"
                },
                {
                    "name": "PT Cruiser"
                },
                {
                    "name": "Saratoga"
                },
                {
                    "name": "Sebring"
                },
                {
                    "name": "Stratus"
                },
                {
                    "name": "Viper"
                },
                {
                    "name": "Vision"
                },
                {
                    "name": "Voyager"
                }
            ]
        },
        {
            "name": "Dodge",
            "models": [
                {
                    "name": "Avenger"
                },
                {
                    "name": "Caliber"
                },
                {
                    "name": "Journey"
                },
                {
                    "name": "Nitro"
                }
            ]
        },
        {
            "name": "Fiat",
            "models": [
                {
                    "name": "124 Spider"
                },
                {
                    "name": "124 Sport/Spider"
                },
                {
                    "name": "126"
                },
                {
                    "name": "127"
                },
                {
                    "name": "131"
                },
                {
                    "name": "500"
                },
                {
                    "name": "500L"
                },
                {
                    "name": "500X"
                },
                {
                    "name": "850 T/900 T/-E"
                },
                {
                    "name": "Albea"
                },
                {
                    "name": "Argenta"
                },
                {
                    "name": "Barchetta"
                },
                {
                    "name": "Brava"
                },
                {
                    "name": "Bravo"
                },
                {
                    "name": "Cinquecento"
                },
                {
                    "name": "Coupe"
                },
                {
                    "name": "Croma"
                },
                {
                    "name": "Doblo"
                },
                {
                    "name": "Ducato"
                },
                {
                    "name": "Fiorino"
                },
                {
                    "name": "Freemont"
                },
                {
                    "name": "Fullback"
                },
                {
                    "name": "Grande Punto"
                },
                {
                    "name": "Idea"
                },
                {
                    "name": "Linea"
                },
                {
                    "name": "Marea"
                },
                {
                    "name": "Multipla"
                },
                {
                    "name": "Palio"
                },
                {
                    "name": "Panda"
                },
                {
                    "name": "Punto"
                },
                {
                    "name": "Qubo"
                },
                {
                    "name": "Regata"
                },
                {
                    "name": "Ritmo"
                },
                {
                    "name": "Scudo"
                },
                {
                    "name": "Sedici"
                },
                {
                    "name": "Seicento"
                },
                {
                    "name": "Stilo"
                },
                {
                    "name": "Strada"
                },
                {
                    "name": "Talento"
                },
                {
                    "name": "Tempra"
                },
                {
                    "name": "Tipo"
                },
                {
                    "name": "Ulysse"
                },
                {
                    "name": "Uno"
                },
                {
                    "name": "X 1/9"
                }
            ]
        },
        {
            "name": "Mercedes-Benz",
            "models": [
                {
                    "name": "107 Coupe/Roadster"
                },
                {
                    "name": "123"
                },
                {
                    "name": "190/190 E"
                },
                {
                    "name": "207 D - 210 D"
                },
                {
                    "name": "A-Klasse"
                },
                {
                    "name": "B-Klasse"
                },
                {
                    "name": "C-Klasse"
                },
                {
                    "name": "CL-Coupe"
                },
                {
                    "name": "CLA-Klasse"
                },
                {
                    "name": "CLC-Klasse"
                },
                {
                    "name": "CLK-Klasse"
                },
                {
                    "name": "CLS-Klasse"
                },
                {
                    "name": "Citan"
                },
                {
                    "name": "SLC-Klasse"
                },
                {
                    "name": "E-Klasse"
                },
                {
                    "name": "E-Klasse T Modell"
                },
                {
                    "name": "G-Modell"
                },
                {
                    "name": "GL-Klasse"
                },
                {
                    "name": "GLA-Klasse"
                },
                {
                    "name": "GLC-Klasse"
                },
                {
                    "name": "GLE-Klasse"
                },
                {
                    "name": "GLK-Klasse"
                },
                {
                    "name": "GLS-Klasse"
                },
                {
                    "name": "GT AMG"
                },
                {
                    "name": "M-Klasse"
                },
                {
                    "name": "MB 100 D"
                },
                {
                    "name": "R-Klasse"
                },
                {
                    "name": "S-Coupe"
                },
                {
                    "name": "S-Klasse"
                },
                {
                    "name": "SL-Klasse"
                },
                {
                    "name": "SLK-Klasse"
                },
                {
                    "name": "SLR McLaren"
                },
                {
                    "name": "SLS AMG"
                },
                {
                    "name": "Sprinter"
                },
                {
                    "name": "Strich Acht"
                },
                {
                    "name": "Transporter"
                },
                {
                    "name": "V-Klasse"
                },
                {
                    "name": "Vaneo"
                },
                {
                    "name": "Viano"
                },
                {
                    "name": "Vito"
                },
                {
                    "name": "Vito Tourer"
                },
                {
                    "name": "X-Klasse"
                }
            ]
        },
        {
            "name": "Porsche",
            "models": [
                {
                    "name": "718 Boxster"
                },
                {
                    "name": "718 Cayman"
                },
                {
                    "name": "911"
                },
                {
                    "name": "918 Spyder"
                },
                {
                    "name": "924"
                },
                {
                    "name": "928"
                },
                {
                    "name": "944"
                },
                {
                    "name": "968"
                },
                {
                    "name": "Boxster"
                },
                {
                    "name": "Cayenne"
                },
                {
                    "name": "Cayman"
                },
                {
                    "name": "Macan"
                },
                {
                    "name": "Panamera"
                },
                {
                    "name": "Panamera Sport Turismo"
                }
            ]
        },
        {
            "name": "Skoda",
            "models": [
                {
                    "name": "Citigo"
                },
                {
                    "name": "Fabia"
                },
                {
                    "name": "Favorit"
                },
                {
                    "name": "Felicia"
                },
                {
                    "name": "Forman"
                },
                {
                    "name": "Karoq"
                },
                {
                    "name": "Kodiaq"
                },
                {
                    "name": "Octavia"
                },
                {
                    "name": "Pickup"
                },
                {
                    "name": "Rapid"
                },
                {
                    "name": "Roomster"
                },
                {
                    "name": "Skoda 105"
                },
                {
                    "name": "Skoda 120"
                },
                {
                    "name": "Skoda 130"
                },
                {
                    "name": "Superb"
                },
                {
                    "name": "Yeti"
                }
            ]
        },
        {
            "name": "Tesla",
            "models": [
                {
                    "name": "Model S"
                },
                {
                    "name": "Model X"
                },
                {
                    "name": "Roadster"
                }
            ]
        },
        {
            "name": "Volkswagen",
            "models": [
                {
                    "name": "Amarok"
                },
                {
                    "name": "Arteon"
                },
                {
                    "name": "Beetle / New Beetle"
                },
                {
                    "name": "Bora"
                },
                {
                    "name": "CC"
                },
                {
                    "name": "Caddy"
                },
                {
                    "name": "Corrado"
                },
                {
                    "name": "Crafter"
                },
                {
                    "name": "Derby"
                },
                {
                    "name": "Eos"
                },
                {
                    "name": "Fox"
                },
                {
                    "name": "Golf I"
                },
                {
                    "name": "Golf II"
                },
                {
                    "name": "Golf III"
                },
                {
                    "name": "Golf IV"
                },
                {
                    "name": "Golf Plus"
                },
                {
                    "name": "Golf V"
                },
                {
                    "name": "Golf VI"
                },
                {
                    "name": "Golf VII"
                },
                {
                    "name": "Golf VII Sportsvan"
                },
                {
                    "name": "Jetta"
                },
                {
                    "name": "Kaefer 1200"
                },
                {
                    "name": "LT"
                },
                {
                    "name": "Lupo"
                },
                {
                    "name": "Passat"
                },
                {
                    "name": "Phaeton"
                },
                {
                    "name": "Polo"
                },
                {
                    "name": "Santana"
                },
                {
                    "name": "Scirocco"
                },
                {
                    "name": "Scirocco II"
                },
                {
                    "name": "Sharan"
                },
                {
                    "name": "T-Roc"
                },
                {
                    "name": "T3"
                },
                {
                    "name": "T4"
                },
                {
                    "name": "T5"
                },
                {
                    "name": "T6"
                },
                {
                    "name": "Taro"
                },
                {
                    "name": "Tiguan"
                },
                {
                    "name": "Tiguan Allspace"
                },
                {
                    "name": "Touareg"
                },
                {
                    "name": "Touran"
                },
                {
                    "name": "Vento"
                }
            ]
        }
    ]
}