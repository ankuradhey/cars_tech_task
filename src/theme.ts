import { DefaultTheme } from "styled-components";

export const defaultTheme: DefaultTheme = {
    borderRadius: "2px",
    palette: {
        common: {
            black: "#4A4A4A",
            white: "#EDEDED",
        },
        primary: {
            main: "#EA7F28",
            contrastText: "#ffffff",
        },
        secondary: {
            main: "#D37324",
            contrastText: "#ffffff",
        },
        tertiary: {
            main: "#ffffff",
            contrastText: "#EDEDED",
        },
    },
    gutterSize: {
        sm: "8px",
        md: "12px",
        lg: "24px",
    },
};
