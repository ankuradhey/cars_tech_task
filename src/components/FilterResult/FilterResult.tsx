import React, { FC, useEffect, useState } from "react";
import { Row, Col } from "react-bootstrap";
import { constant } from "../../constant";
import { Car, FilterResultProps } from "../../types";
import { CarCard } from "../CarCard";
import { LoadingCarCard } from "../Loader/LoadingCard";
import { Paginator } from "../Paginator/Paginator";
import { Text } from "../Text";

export const FilterResult: FC<FilterResultProps> = ({ color, manufacturer }) => {
    const [cars, setCars] = useState(new Array(10).fill(null));
    const [page, setPage] = useState(1);
    const [carsData, setCarsData] = useState({});
    const [loading, setLoading] = useState(true);

    const navigateToPage = (page) => {
        setPage(page);
    };

    useEffect(() => {
        setLoading(true);
        const fetchData = async (color, manufacturer) => {
            let url = `${constant.url}?page=${page}`;

            // If color is found then add to url string
            if (color) {
                url += `&color=${color}`;
            }
            // If manufacturer is found then add to url string
            if (manufacturer) {
                url += `&manufacturer=${manufacturer}`;
            }

            const result = await fetch(`${url}`);
            const data = await result.json();
            setCars(data.cars);
            setCarsData({ totalPages: data.totalPageCount, totalRecords: data.totalCarsCount });
            setLoading(false);
        };
        fetchData(color, manufacturer);
    }, [page, color, manufacturer]);

    // Another useeffect to set first page when fresh record appears
    useEffect(() => {
        setPage(1);
    }, [color, manufacturer]);

    return (
        <>
            <Text size="large" isBold className="d-block margin-bottom-2">
                Available cars
            </Text>
            <Text size="large" className="margin-bottom-3">
                {loading ? "Loading" : `Showing ${cars.length} of ${carsData.totalRecords} results`}
            </Text>

            {cars.map((car: Car) => {
                if (loading) return <LoadingCarCard />;

                const title = `${car?.manufacturerName} ${car?.modelName}`;
                const subTitle = `Stock # ${car?.stockNumber}-${
                    car?.mileage?.number
                }${car?.mileage?.unit.toUpperCase()}-${car?.fuelType}-${car?.color}`;

                return (
                    <Row>
                        <Col>
                            <CarCard
                                title={title}
                                subTitle={subTitle}
                                id={car?.stockNumber}
                                pictureUrl={car?.pictureUrl}
                            />
                        </Col>
                    </Row>
                );
            })}
            <Paginator
                totalPages={carsData.totalPages}
                currentPage={page}
                navigateToPage={navigateToPage}
            />
        </>
    );
};
