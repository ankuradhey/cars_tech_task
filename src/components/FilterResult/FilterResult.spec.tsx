import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import { Router, Route, MemoryRouter } from "react-router-dom";
import { createMemoryHistory } from "history";
import nock from "nock";
import { FilterResult } from "./FilterResult";
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../../theme";
import * as carsMock from "../../mock/cars.json";
import * as colorsMock from "../../mock/colors.json";
import * as manufacturersMock from "../../mock/manufacturers.json";

import { constant } from "../../constant";

describe("FilterResult Component", () => {
    let history, component;

    beforeEach(() => {
        nock(`${constant.host}`)
            .defaultReplyHeaders({
                "access-control-allow-origin": "*",
                "access-control-allow-credentials": "true",
            })
            .get(`${constant.carsUrl}?page=1`)
            .reply(200, carsMock);

        nock(`${constant.host}`)
            .defaultReplyHeaders({
                "access-control-allow-origin": "*",
                "access-control-allow-credentials": "true",
            })
            .get(constant.colorsUrl)
            .reply(200, colorsMock);

        nock(`${constant.host}`)
            .defaultReplyHeaders({
                "access-control-allow-origin": "*",
                "access-control-allow-credentials": "true",
            })
            .get(constant.manufacturersUrl)
            .reply(200, manufacturersMock);

        history = createMemoryHistory();

        component = render(
            <ThemeProvider theme={defaultTheme}>
                <MemoryRouter initialEntries={["/"]}>
                    <FilterResult />
                </MemoryRouter>
            </ThemeProvider>
        );
    });

    it("Should do full app rendering/navigating", async () => {
        const { container } = component;
        expect(screen.getByText(/Available cars/i)).toBeInTheDocument();
        await waitFor(() => screen.findByText("Showing 10 of 1000 results"));
        expect(container).toMatchSnapshot();
    });

    it("Should display pagination items", async () => {
        await waitFor(() => screen.findByText("Showing 10 of 1000 results"));
        expect(screen.getByText(/First/i)).toBeInTheDocument();
        expect(screen.getByText(/Previous/i)).toBeInTheDocument();
        expect(screen.getByText(/Next/i)).toBeInTheDocument();
        expect(screen.getByText(/Last/i)).toBeInTheDocument();
        expect(screen.getByText(/Page 1 of 10/i)).toBeInTheDocument();
    });

    it("Should display car list", async () => {
        await waitFor(() => screen.findByText("Showing 10 of 1000 results"));
        expect(
            screen.getByText(`${carsMock.cars[0].manufacturerName} ${carsMock.cars[0].modelName}`)
        ).toBeInTheDocument();
    });
});
