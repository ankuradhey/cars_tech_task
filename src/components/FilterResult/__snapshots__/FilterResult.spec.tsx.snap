// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterResult Component Should do full app rendering/navigating 1`] = `
<div>
  <div
    class="sc-dlfnbm iPsSGC d-block margin-bottom-2"
  >
    Available cars
  </div>
  <div
    class="sc-dlfnbm dqWGDI margin-bottom-3"
  >
    Showing 10 of 1000 results
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ dQCTDP"
              height="80px"
              loading="false"
              width="100px"
            >
              <img
                alt="car"
                src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
              />
            </div>
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="card-title h5"
            >
              Fiat Idea
            </div>
            <div
              class="sc-dlfnbm dXPfWA"
            >
              Stock # 10019 - 111955KM - Petrol - blue
            </div>
            <button
              class="sc-gsTCUz ezYRfg d-block"
              data-testid="viewDetailsLink"
            >
              View details
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ dQCTDP"
              height="80px"
              loading="false"
              width="100px"
            >
              <img
                alt="car"
                src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
              />
            </div>
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="card-title h5"
            >
              Skoda Rapid
            </div>
            <div
              class="sc-dlfnbm dXPfWA"
            >
              Stock # 10084 - 166819KM - Petrol - white
            </div>
            <button
              class="sc-gsTCUz ezYRfg d-block"
              data-testid="viewDetailsLink"
            >
              View details
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ dQCTDP"
              height="80px"
              loading="false"
              width="100px"
            >
              <img
                alt="car"
                src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
              />
            </div>
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="card-title h5"
            >
              Mercedes-Benz Sprinter
            </div>
            <div
              class="sc-dlfnbm dXPfWA"
            >
              Stock # 10176 - 125331KM - Petrol - silver
            </div>
            <button
              class="sc-gsTCUz ezYRfg d-block"
              data-testid="viewDetailsLink"
            >
              View details
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ dQCTDP"
              height="80px"
              loading="false"
              width="100px"
            >
              <img
                alt="car"
                src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
              />
            </div>
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="card-title h5"
            >
              Mercedes-Benz Viano
            </div>
            <div
              class="sc-dlfnbm dXPfWA"
            >
              Stock # 10217 - 119173KM - Diesel - green
            </div>
            <button
              class="sc-gsTCUz ezYRfg d-block"
              data-testid="viewDetailsLink"
            >
              View details
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ dQCTDP"
              height="80px"
              loading="false"
              width="100px"
            >
              <img
                alt="car"
                src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
              />
            </div>
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="card-title h5"
            >
              Mercedes-Benz SL-Klasse
            </div>
            <div
              class="sc-dlfnbm dXPfWA"
            >
              Stock # 10239 - 165176KM - Petrol - blue
            </div>
            <button
              class="sc-gsTCUz ezYRfg d-block"
              data-testid="viewDetailsLink"
            >
              View details
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ dQCTDP"
              height="80px"
              loading="false"
              width="100px"
            >
              <img
                alt="car"
                src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
              />
            </div>
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="card-title h5"
            >
              Chrysler Crossfire
            </div>
            <div
              class="sc-dlfnbm dXPfWA"
            >
              Stock # 10283 - 114177KM - Petrol - black
            </div>
            <button
              class="sc-gsTCUz ezYRfg d-block"
              data-testid="viewDetailsLink"
            >
              View details
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ dQCTDP"
              height="80px"
              loading="false"
              width="100px"
            >
              <img
                alt="car"
                src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
              />
            </div>
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="card-title h5"
            >
              Porsche 718 Cayman
            </div>
            <div
              class="sc-dlfnbm dXPfWA"
            >
              Stock # 10385 - 180397KM - Petrol - white
            </div>
            <button
              class="sc-gsTCUz ezYRfg d-block"
              data-testid="viewDetailsLink"
            >
              View details
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ dQCTDP"
              height="80px"
              loading="false"
              width="100px"
            >
              <img
                alt="car"
                src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
              />
            </div>
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="card-title h5"
            >
              BMW i3
            </div>
            <div
              class="sc-dlfnbm dXPfWA"
            >
              Stock # 10477 - 127584KM - Petrol - blue
            </div>
            <button
              class="sc-gsTCUz ezYRfg d-block"
              data-testid="viewDetailsLink"
            >
              View details
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ dQCTDP"
              height="80px"
              loading="false"
              width="100px"
            >
              <img
                alt="car"
                src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
              />
            </div>
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="card-title h5"
            >
              Volkswagen Crafter
            </div>
            <div
              class="sc-dlfnbm dXPfWA"
            >
              Stock # 10481 - 176629KM - Diesel - white
            </div>
            <button
              class="sc-gsTCUz ezYRfg d-block"
              data-testid="viewDetailsLink"
            >
              View details
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="margin-bottom-2 card"
      >
        <div
          class="sc-hKgILt hZGhHr"
        >
          <div
            class="sc-eCssSg ituOTJ"
          >
            <div
              class="sc-bdfBwQ dQCTDP"
              height="80px"
              loading="false"
              width="100px"
            >
              <img
                alt="car"
                src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
              />
            </div>
          </div>
          <div
            class="sc-jSgupP gstVBl"
          >
            <div
              class="card-title h5"
            >
              Dodge Nitro
            </div>
            <div
              class="sc-dlfnbm dXPfWA"
            >
              Stock # 10571 - 115196KM - Petrol - yellow
            </div>
            <button
              class="sc-gsTCUz ezYRfg d-block"
              data-testid="viewDetailsLink"
            >
              View details
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="d-flex justify-content-center"
  >
    <div
      class="p-2 bd-highlight"
    >
      <div
        class="sc-dlfnbm icINku"
        data-testid="first-page"
      >
        First
      </div>
    </div>
    <div
      class="p-2 bd-highlight"
    >
      <div
        class="sc-dlfnbm icINku"
        data-testid="previous-page"
      >
        Previous
      </div>
    </div>
    <div
      class="p-2 bd-highlight"
    >
      <div
        class="sc-dlfnbm icINku"
      >
        Page 
        1
         of 
        100
      </div>
    </div>
    <div
      class="p-2 bd-highlight"
    >
      <button
        class="sc-gsTCUz ezYRfg"
        data-testid="next-page"
      >
        Next
      </button>
    </div>
    <div
      class="p-2 bd-highlight"
    >
      <button
        class="sc-gsTCUz ezYRfg"
        data-testid="last-page"
      >
        Last
      </button>
    </div>
  </div>
</div>
`;
